code/game/machinery/bank_machine.dm

```diff
@@ -95,14 +87,57 @@
 	switch(action)
 		if("siphon")
 			say("Siphon of station credits has begun!")
-			siphoning = TRUE
+			start_siphon()
 			. = TRUE
 		if("halt")
 			say("Station credit withdrawal halted.")
-			end_syphon()
+			end_siphon()
 			. = TRUE
 
-/obj/machinery/computer/bank_machine/proc/end_syphon()
+/obj/machinery/computer/bank_machine/proc/start_siphon()
+	siphoning = TRUE
+
+/obj/machinery/computer/bank_machine/proc/end_siphon()
 	siphoning = FALSE
-	new /obj/item/holochip(drop_location(), syphoning_credits) //get the loot
-	syphoning_credits = 0
+	new /obj/item/holochip(drop_location(), siphoning_credits) //get the loot
+	siphoning_credits = 0
```

Comment:
should do early return when `if(!siphoning || !siphoning_credits)` because we don't want to make zero credit item.